### ISSUE TITLE
Make region import robust even with messy polygons

### ISSUE THREAD

**Issue by kring** [TerriaJS/magda-metadata#6]
Region polygons often have problems, such as:
* Unclosed linear rings (the first and last point are not the same, even though GeoJSON says they must be).
* Duplicate positions in a linear ring, other than the first and last.  This indicates a loop in the ring, which is not allowed by the OGC simple feature specification, even though it is allowed and common and Esri shapefiles.  Elasticsearch seems to expect GeoJSON polygons to conform to the simple feature specification.
* Self-intersecting segments in the polygon.

In a perfect world, whoever provided the regions would fix these problems.  But in reality our system needs to be robust in the face of imperfect data.  To that end, we should automatically clean up these problems in the regions we load.  This recent paper, _A triangulation-based approach to automatically repair GIS polygons_, proposes a very promising approach:
http://dx.doi.org/10.1016/j.cageo.2014.01.009
http://www.sciencedirect.com/science/article/pii/S009830041400020X

There is an implementation of this approach, but it is not suitable for use in Magda because it uses the GPL licence.  It should be straightforward to implement it ourselves, though.

**Comment by steve9164** [comment]
This is how I fixed the invalid geometries I had (for region mapping files):
```
ogr2ogr -f GeoJSON fixed.json invalid_geometries.json -dialect SQLITE -sql "SELECT ST_MakeValid(geometry) as geometry, * FROM OGRGeoJSON"
```

Would that help? @kring @AlexGilleran

**Comment by kring** [comment]
Yes, that's great!  I didn't realize you can use ogr2ogr to execute `ST_MakeValid`.  For what it's worth, the paper above compares itself to MakeValid.  The paper's approach has some advantages, but this is still great.
